modules/data.atmosphere/R/GEFS_helper_functions.R

```diff
@@ -0,0 +1,906 @@
+        dplyr::filter(!is.na(air_temperature))
+      
+      end_date <- forecast_noaa_ens %>%
+        dplyr::summarise(max_time = max(time))
+      
+      results = data.frame(
+        file = "",                            #Path to the file (added in loop below).
+        host = PEcAn.remote::fqdn(),          #Name of the server where the file is stored
+        mimetype = "application/x-netcdf",    #Format the data is saved in
+        formatname = "CF Meteorology",        #Type of data
+        startdate = paste0(format(forecast_date, "%Y-%m-%dT%H:%M:00")),    #starting date and time, down to the second
+        enddate = paste0(format(end_date$max_time, "%Y-%m-%dT%H:%M:00")),        #ending date and time, down to the second
+        dbfile.name = "NOAA_GEFS_downscale",            #Source of data (ensemble number will be added later)
+        stringsAsFactors = FALSE
+      )
+      
+      identifier = paste("NOAA_GEFS", site_id, ens_name, format(forecast_date, "%Y-%m-%dT%H:%M"), 
+                         format(end_date$max_time, "%Y-%m-%dT%H:%M"), sep="_")     
+      
+      fname <- paste0(identifier, ".nc")
+      ensemble_folder = file.path(output_directory, identifier)
+      output_file <- file.path(ensemble_folder,fname)
+      
+      if (!dir.exists(ensemble_folder)) {
+        dir.create(ensemble_folder, recursive=TRUE, showWarnings = FALSE)} 
+      
+      
+      #Write netCDF
+      noaaGEFSpoint::write_noaa_gefs_netcdf(df = forecast_noaa_ens,ens, lat = lat_list[1], lon = lon_east, cf_units = cf_var_units1, output_file = output_file, overwrite = TRUE)
+      
+      if(downscale){
+        #Downscale the forecast from 6hr to 1hr
+        
+        
+        identifier_ds = paste("NOAA_GEFS_downscale", site_id, ens_name, format(forecast_date, "%Y-%m-%dT%H:%M"), 
+                              format(end_date$max_time, "%Y-%m-%dT%H:%M"), sep="_")
+        
+        fname_ds <- paste0(identifier_ds, ".nc")
+        ensemble_folder_ds = file.path(output_directory, identifier_ds)
+        output_file_ds <- file.path(ensemble_folder_ds,fname_ds)
+        
+        if (!dir.exists(ensemble_folder_ds)) {
+          dir.create(ensemble_folder_ds, recursive=TRUE, showWarnings = FALSE)} 
+        
+        results$file = output_file_ds
+        results$dbfile.name = fname_ds
+        results_list[[ens]] <- results
+        
+        #Run downscaling
+        noaaGEFSpoint::temporal_downscale(input_file = output_file, output_file = output_file_ds, overwrite = TRUE, hr = 1)
+      }
+      
+      
+    }
+  }
+  return(results_list)
+} #process_gridded_noaa_download
+
+#' @title Downscale NOAA GEFS frin 6hr to 1hr
+#' @return None
+#'
+#' @param input_file, full path to 6hr file
+#' @param output_file, full path to 1hr file that will be generated
+#' @param overwrite, logical stating to overwrite any existing output_file
+#' @param hr time step in hours of temporal downscaling (default = 1)
+#' @export
+#'
+#' @author Quinn Thomas
```

Comment:
If there's code that Quinn wrote in another repo, and we're just copying that code, I'd ask whether there's a way to point to his code directly (e.g. load it as a package). Otherwise the codes will diverge through time and we'll miss (potentially important) updates.

Reply:
I've changed a small part of most of the functions to get it to work within the framework we already had set up with GEFS and Pecan